hearthstone/user_agent.py

```diff
@@ -131,7 +131,22 @@ def parse_buy_input(user_input: str, player: 'Player') -> Optional[Action]:
         elif split_list[0] == "u":
             return TavernUpgradeAction()
         elif split_list[0] == "h":
-            return HeroPowerAction()
+            if not 0 < len(split_list) <= 2:
```

Comment:
The usual convention of range checking is inclusive lower bound, and exclusive upper bound: http://www.cs.utexas.edu/users/EWD/transcriptions/EWD08xx/EWD831.html

Reply:
Good to know!